src/Paket.Core/embedded/Paket.Restore.targets

```diff
@@ -69,9 +69,11 @@
       <PaketRestoreRequired Condition=" '$(PaketRestoreLockFileHash)' == '' ">true</PaketRestoreRequired>
     </PropertyGroup>
 
+
     <!-- Do a global restore if required -->
     <Exec Command='$(PaketBootStrapperCommand)' Condition="Exists('$(PaketBootStrapperExePath)') AND !(Exists('$(PaketExePath)'))" ContinueOnError="false" />
-    <Exec Command='$(PaketCommand) restore' Condition=" '$(PaketRestoreRequired)' == 'true' " ContinueOnError="false" />
```

Comment:
Note that calling `paket restore` is by design here. We tweaked `paket restore` for maximum performance (including skipped command line parsing). With this change restore will most likely be a lot slower if stuff was already restored.

Reply:
What I'm trying to say is that we either need a different solution / carefully tweak performance for this case or some other trick